### DISCUSSION TITLE
Deployments

### DISCUSSION THREAD

**Discussion by AbhijithGanesh** [CiviWiki/OpenCiviWiki#1002] [Ideas]
I would like to know how this project is deployed and I would like to help the devops team if there is a requirement.

**Comment by brylie** [comment]
The project isn't currently being deployed, since we are in a significant code transition. We will need to consider deployment though. What are your thoughts on deployment options?

Bear in mind that deployment should be very simple, mostly automated, and be compatible with generic VPS/PaaS providers.

For what it's worth, I would recommend we use one of the following PaaS solutions

- [Dokku](https://dokku.com/)
- [DigitalOcean App Platform](https://www.digitalocean.com/products/app-platform/)

**Comment by gorkemarslan** [comment]
I have previous experience in DigitalOcean (Paas) and AWS EC2 (IaaS). I've not used Dokku before. 

I know why you want to use PaaS: because of its simplicity. :) With IaaS, there is tons of stuff to configure and some efforts are needed to solve problems. So PaaS is a good choice.

The good thing is for DigitalOcean is that they have really great documents and blog posts. Their supports on users' issues and problems are counted as good. 

Before making of choice between Dokku and DigitalOcean, we need to search through some tutorials for CI/ CD pipelines for them to learn how difficult or simple to fully automate the whole process.

Do you have any idea about which CI service we are going to use? (GitHub actions, CircleCI, etc.)

**Reply by AbhijithGanesh** [reply]
I'd rather prefer GitHub auctions for its reliability and ease to integrate

**Reply by brylie** [reply]
Agreed. DigitalOcean App Platform seems like the best candidate for simplicity and pricing. Along with the great documentation, you mention, they are also strong supporters of open-source (two additional points in their favor).

Regarding CI, I don't have an opinion. Perhaps we can open a separate CI thread to discuss and vote on the issue? That, or we can just agree here to use GitHub actions for now.

**Reply by AbhijithGanesh** [reply]
Does DigitalOcean not support Automatic deployment ? Heroku does when a branch receives changes, isn't this feature available on DigitalOcean?

**Reply by brylie** [reply]
> Does DigitalOcean not support Automatic deployment ?

DigitalOcean App Platform supports automatic deployment from GitHub branches, as can be seen in their [Quickstart Guide](https://docs.digitalocean.com/products/app-platform/quickstart/#create-an-app).

**Reply by AbhijithGanesh** [reply]
I am in favour of digitalocean then

**Comment by AbhijithGanesh** [comment]
@gorkemarslan and @Brylie is there any change required from my end to merge v2Develop into Develop

**Reply by AbhijithGanesh** [reply]
If it is registered for Hacktoberfest, then would that mean it'll be kept on hold for another day or so?

**Reply by gorkemarslan** [reply]
Could you write this in #994? It can be more proper. :) And could you delete these messages?  We can continue conversations in the right places. :D
